plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2727,6 +2728,72 @@ public void testLocalDynamicFilteringWithSelectiveBuildSizeJoin()
         });
     }
 
+    @Test(dataProvider = "repartitioningDataProvider")
+    public void testRepartitionDataOnCtas(String partitioning, int expectedFiles)
+    {
+        testRepartitionData(true, partitioning, expectedFiles);
+    }
+
+    @Test(dataProvider = "repartitioningDataProvider")
+    public void testRepartitionDataOnInsert(String partitioning, int expectedFiles)
+    {
+        testRepartitionData(false, partitioning, expectedFiles);
+    }
+
+    private void testRepartitionData(boolean ctas, String partitioning, int expectedFiles)
+    {
+        String tableName = "repartition_" +
+                (ctas ? "ctas" : "insert") +
+                "_" + partitioning.replaceAll("[^a-zA-Z0-9]", "") +
+                "_" + randomTableSuffix();
+
+        long rowCount = (long) computeScalar("SELECT count(*) FROM orders");
+
+        if (ctas) {
+            assertUpdate(
```

Comment:
We could use the below properties to ensure that there is no influence of DeterminePreferredWritePartitioning or the default configs here.
```
private Session withForcedPreferredPartitioning()
    {
        return Session.builder(getQueryRunner().getDefaultSession())
                .setSystemProperty(USE_PREFERRED_WRITE_PARTITIONING, "true")
                .setSystemProperty(PREFERRED_WRITE_PARTITIONING_MIN_NUMBER_OF_PARTITIONS, "1")
                .build();
    }
```

Reply:
I am aware of `DeterminePreferredWritePartitioning`.
Actually, it has problem that it assumes simple partitioing schemes.
For example, i can partition on first letter in iceberg (`truncate(name,1)`). `DeterminePreferredWritePartitioning` will look at cardinality of the source column, and it doesn't know the cardinality of the projection we're calculating. 
Not sure if this matters much. (-> https://github.com/trinodb/trino/issues/9852)

anyway, here i wanted to test with defaults. @raunaqmorarka  do you think i should control `DeterminePreferredWritePartitioning` behavior tightly?

cc @losipiuk

Reply:
added